### DISCUSSION TITLE
access existing store data on SSR

### DISCUSSION THREAD

**Discussion by dellwatson** [pmndrs/zustand#1788] [General]
following this https://github.com/pmndrs/zustand/issues/182
i dont see there's any improvement here yet? 

i tried load my zustand store SSR and it shows me empty, though i did load it already on previous page. if i move it into inside component, it shows me the data but not in serversideprops, it shows me the initial data, and i want to avoid load it again as there's some complexity..

```js
export const getServerSideProps: GetServerSideProps<DetailCourseProps> = async (
  context
) => {

  const { id, course_id, timeplay, order_index } = context.query;

  // Load courses from the store using the `loadCourses` method
  const { courses }: any = useCoursesStore.getState();


  const c = courses.filter((item: any) => {
    return item?.category_id === id;
  });
  const result = c.sort((a: any, b: any) => a.order - b.order);

  return {
    props: {
      data: result,
      id: id as string,

    },
  };
};

```

**Comment by dai-shi** [comment]
Do you mean you want shared data across sessions? It would be possible to do with zustand global store, but it would be better to do with simpler caching solution, because it doesn't use most of zustand features and it's confusing.

**Reply by dellwatson** [reply]
the zustand global store gives me initial value when accessing on server `getServersideProps`, but on client it is ok

1. on page A: set zustand store
2. navigating to page B
3. serverside page B trying to get state that been loaded from step1 `const { courses }: any = useCoursesStore.getState();` gives empty / initial value
4. client side page B loaded, and trying to get state from step 1 and it works (but i dont want client load)

i end up using batch cookies on step1 and on step 3 serverside load the cookies

**Reply by dai-shi** [reply]
Oh, it's a very different story from what I expected in my previous comment.
It's an open problem. That is, how to coordinate client state into server state.
Libraries and frameworks (like next.js) often use url state, and cookies are fine too.

(fwiw, that's my next big challenge in the context of react server components.)
